notebooks/grism_efficency_vs_fringe_frequency.ipynb

```diff
@@ -0,0 +1,1399 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Grism Efficiency vs Fringe Frequency\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# external\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import plot\n",
+    "from astropy.units import unit\n",
+    "\n",
+    "# project\n",
+    "from payload_designer.components import diffractors"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "Run this section onwards when a parameter is updated."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Parameters"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 39,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# region constants\n",
+    "\n",
+    "a = 90\n",
+    "d = 2.5\n",
+    "eff_mat = 0.85\n",
+    "m = 1\n",
+    "n_1 = 1.0\n",
+    "n_2 = 1.52\n",
+    "n_3 = 1.3\n",
+    "n_g = 0.1\n",
```

Comment:
One of the things we've started moving towards is the use of descriptive variable names. Especially for very complicated functions, it can be hard to tell at a glance the meaning of each variable. Let's make use of descriptive variables names here (i.e. `l` -> `wavelength`, `a_in` -> `incident_angle`, etc). This also prevents us from having to flip back and forth between the MCR document or check the function docstrings to look up variable definitions everytime.